contracts/sentinel/guards/MaxFeiWithdrawalGuard.sol

```diff
@@ -0,0 +1,120 @@
+    PCVGuardian public constant pcvGuardian = PCVGuardian(0x02435948F84d7465FB71dE45ABa6098Fc6eC2993);
+
+    /// @notice the minimum amount of underlying which can be withdrawn from a deposit that registers in the guard.
+    /// i.e. if the min is 100 FEI but the amount in the contract is 1 FEI, the amountToWithdraw will return 0 and the check will fail
+    /// @dev added to prevent dust from bricking the contract
+    uint256 public constant MIN_WITHDRAW = 100e18;
+
+    constructor(
+        address core,
+        address[] memory deposits,
+        address[] memory destinations,
+        address[] memory liquiditySources
+    ) CoreRef(core) {
+        uint256 len = deposits.length;
+        require(len == destinations.length);
+        for (uint256 i = 0; i < len; ) {
+            depositSet.add(deposits[i]);
+            withdrawInfos[deposits[i]] = WithdrawInfo({
+                destination: destinations[i],
+                liquiditySource: liquiditySources[i]
+            });
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// @notice setter for the deposit destination and liquidity source
+    function setWithdrawInfo(address deposit, WithdrawInfo calldata withdrawInfo)
+        public
+        hasAnyOfThreeRoles(TribeRoles.GUARDIAN, TribeRoles.GOVERNOR, TribeRoles.PCV_SAFE_MOVER_ROLE)
+    {
+        withdrawInfos[deposit] = withdrawInfo;
+    }
+
+    /// @notice check if contract can be called. If any deposit has a nonzero withdraw amount available, then return true.
+    function check() external view override returns (bool) {
+        for (uint256 i = 0; i < depositSet.length(); ) {
+            if (getAmountToWithdraw(IPCVDeposit(depositSet.at(i))) > 0) return true;
+            unchecked {
+                ++i;
+            }
+        }
+        return false;
+    }
+
+    /// @notice return the amount that can be withdrawn from a deposit after leaving min liquidity
+    function getAmountToWithdraw(IPCVDeposit deposit) public view returns (uint256) {
+        // Reserves of underlying left in the liquidity source are considered withdrawable liquidity
+        uint256 liquidity = fei().balanceOf(withdrawInfos[address(deposit)].liquiditySource);
+        if (liquidity == 0) {
+            return 0;
+        }
+
+        // max withdraw is the pcv deposit balance
+        uint256 withdrawAmount = deposit.balance();
+        if (withdrawAmount > liquidity) {
+            withdrawAmount = liquidity;
+        }
+        return withdrawAmount > MIN_WITHDRAW ? withdrawAmount : 0;
+    }
+
+    /// @notice return the first element which can be withdrawn from with the appropriate calldata encoding tha max withdraw amount.
+    /// @dev it only returns one element for simplicity. Unlikely multiple will be withdrawable simulteneously after the first pass, and pruning empty entries from a sparse array is an inefficient and inelegant algorithm in solidity.
```

Comment:
I can envision a potential accidental self-dos here if we accidentally create a circular loop of withdrawing from one deposit into another. I don't think there's a high chance of this happening, but we should at least be aware of the possibility.

Reply:
(then we'd not be able to withdraw from any other deposits!)